doozer/doozerlib/assembly_inspector.py

```diff
@@ -21,42 +21,47 @@
-                  AssemblyIssues, fetch rhcos_builds and other defined methods
-            "images": Do the lookups to enable image inspection, but expect code touching group RPMs
-                      to fail (limited use case)
-            "both": Do the lookups for a full inspection
         """
         self.runtime = runtime
         self.brew_session = brew_session
-        if lookup_mode and runtime.mode != lookup_mode:
-            raise ValueError(f'Runtime must be initialized with "{lookup_mode}"')
 
         self.assembly_rhcos_config = assembly_rhcos_config(self.runtime.releases_config, self.runtime.assembly)
         self.assembly_type: AssemblyTypes = assembly_type(self.runtime.releases_config, self.runtime.assembly)
         self._rpm_build_cache: Dict[int, Dict[str, Optional[Dict]]] = {}  # Dict[rhel_ver] -> Dict[distgit_key] -> Optional[BuildDict]
         self._external_rpm_build_cache: Dict[str, Dict[str, Optional[Dict]]] = {}  # Dict[tag] -> Dict[distgit_key] -> Optional[BuildDict]
         self._permits = assembly_permits(self.runtime.releases_config, self.runtime.assembly)
+        self._rpm_deliveries: Dict[str, RPMDelivery] = {}  # Dict[package_name] => per package RpmDelivery config
+        self._release_build_record_inspectors: Dict[str, Optional[BuildRecordInspector]] = dict()
+
+    async def initialize(self, lookup_mode: Optional[str] = "both"):
+        """
+        :param lookup_mode:
+            None: Create a lite version without the ability to inspect Images; can be used to check
+                  AssemblyIssues, fetch rhcos_builds and other defined methods
+            "images": Do the lookups to enable image inspection, but expect code touching group RPMs
+                      to fail (limited use case)
+            "both": Do the lookups for a full inspection
+        """
+        if lookup_mode and self.runtime.mode != lookup_mode:
+            raise ValueError(f'Runtime must be initialized with "{lookup_mode}"')
 
         if not lookup_mode:  # do no lookups
             return
         # If an image component has a latest build, an ImageInspector associated with the image.
-        self._release_image_inspectors: Dict[str, Optional[BrewBuildImageInspector]] = dict()
-        for image_meta in runtime.get_for_release_image_metas():
-            latest_build_obj = image_meta.get_latest_brew_build(default=None, el_target=image_meta.branch_el_target())
+        for image_meta in self.runtime.get_for_release_image_metas():
+            latest_build_obj = await image_meta.get_latest_build(default=None, el_target=image_meta.branch_el_target())
             if latest_build_obj:
-                self._release_image_inspectors[image_meta.distgit_key] = BrewBuildImageInspector(self.runtime, latest_build_obj['nvr'])
+                build_record_inspector_cls = BuildRecordInspector.get_build_record_inspector_cls(self.runtime)
```

Comment:
we can take this out of the loop?

Reply:
I am not sure I get what you mean?

Reply:
Maybe this? f5dcabaa41ffecdc40588b751afccdab755499af